samples/machine-learning/parallel-half-moons/Host.cs

```diff
@@ -0,0 +1,157 @@
+// Licensed under the MIT License.
+
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Simulators;
+using System;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Runtime.InteropServices;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using static System.Math;
+
+namespace Microsoft.Quantum.Samples
+{
+    using Microsoft.Quantum.MachineLearning;
+
+    class Program
+    {
+        static async Task Main()
+        {
+            // We start by loading the training and validation data from our JSON
+            // data file.
+            var data = await LoadData("data.json");
+
+            // We then define the classifier parameters where we want to start our
+            // training iterations from. Since gradient descent is good at finding
+            // local optima, it's helpful to have a variety of different starting
+            // points.
+            var parameterStartingPoints = new []
+            {
+                new [] {0.060057, 3.00522,  2.03083,  0.63527,  1.03771, 1.27881, 4.10186,  5.34396},
+                new [] {0.586514, 3.371623, 0.860791, 2.92517,  1.14616, 2.99776, 2.26505,  5.62137},
+                new [] {1.69704,  1.13912,  2.3595,   4.037552, 1.63698, 1.27549, 0.328671, 0.302282},
+                new [] {5.21662,  6.04363,  0.224184, 1.53913,  1.64524, 4.79508, 1.49742,  1.5455}
+            };
+
+            // Convert samples to Q# form.
+            var samples = new QArray<QArray<double>>(data.TrainingData.Features.Select(vector => new QArray<double>(vector)));
+
+            // Once we have the data loaded and have initialized our target machine,
+            // we can then use that target machine to train a QCC classifier.
+            var (optimizedParameters, optimizedBias, nMisses) = parameterStartingPoints
+                // We can use parallel LINQ (PLINQ) to convert the IEnumerable
+                // over starting points into a parallelized query.
+                .AsParallel()
+                // By default, PLINQ may or may not actually run our query in
+                // parallel, depending on the capabilities of your machine.
+                // We can force PLINQ to actually parallelize, however, by using
+                // the WithExecutionMode method.
+                .WithExecutionMode(ParallelExecutionMode.ForceParallelism)
+                // Many of the same LINQ methods are defined for PLINQ queries
+                // as well as IEnumerable objects, so we can go on and run
+                // the training loop in parallel by selecting on the start point.
+                .Select(
+                    (startPoint, idxStartPoint) =>
+                    {
+                        // Since we want each start point to run on its own
+                        // instance of the full-state simulator, we create a new
+                        // instance here, using C# 8's "using var" syntax to
```

Comment:
```suggestion
                        // instance here, using C# 8's "using" syntax to
```
You can actually use `using` this way with an explicit type, not just `var`.

Reply:
Good point, thank you! As with @geduardo's feedback, I'll defer to the next PR so that I can close this protected branch.